core/src/utilities/helpers/routing-helpers.js

```diff
@@ -0,0 +1,196 @@
+  } else if (children && children.length > 0) {
+    return children[0].pathSegment;
+  } else {
+    return '';
+  }
+};
+
+export const isExistingRoute = function(path, pathData) {
+  if (!path) {
+    return true;
+  }
+
+  const lastElement =
+    pathData.navigationPath[pathData.navigationPath.length - 1];
+  const routeSplit = path.replace(/\/$/, '').split('/');
+  const lastPathSegment = routeSplit[routeSplit.length - 1];
+
+  return lastElement.pathSegment === lastPathSegment;
+};
+
+export const hideElementChildren = node => {
+  if (node.children) {
+    Array.from(node.children).forEach(child => {
+      child.style.display = 'none';
+    });
+  }
+};
+
+export const removeElementChildren = node => {
+  while (node.firstChild) {
+    node.removeChild(node.firstChild);
+  }
+};
+
+export const getLocation = url => {
+  var element = document.createElement('a');
+  element.href = url;
+  return element.origin;
+};
+
+export const parseParams = paramsString => {
+  const result = {};
+  const viewParamString = paramsString;
+  const pairs = viewParamString ? viewParamString.split('&') : null;
+  if (pairs) {
+    pairs.forEach(pairString => {
+      const keyValue = pairString.split('=');
+      if (keyValue && keyValue.length > 0) {
+        result[decodeURIComponent(keyValue[0])] = decodeURIComponent(
+          keyValue[1]
+        );
+      }
+    });
+  }
+  return result;
+};
+
+export const replaceVars = (viewUrl, params, prefix, parenthesis = true) => {
+  let processedUrl = viewUrl;
+  if (params) {
+    Object.entries(params).forEach(entry => {
+      processedUrl = processedUrl.replace(
+        new RegExp(
+          escapeRegExp(
+            (parenthesis ? '{' : '') +
+              prefix +
+              entry[0] +
+              (parenthesis ? '}' : '')
+          ),
+          'g'
+        ),
+        encodeURIComponent(entry[1])
+      );
+    });
+  }
+  if (parenthesis) {
+    processedUrl = processedUrl.replace(
+      new RegExp('\\{' + escapeRegExp(prefix) + '[^\\}]+\\}', 'g'),
+      ''
+    );
+  }
+  return processedUrl;
+};
+
+export const getNodeParams = params => {
+  const result = {};
+  const paramPrefix = getContentViewParamPrefix();
+  if (params) {
+    Object.entries(params).forEach(entry => {
+      if (entry[0].startsWith(paramPrefix)) {
+        const paramName = entry[0].substr(paramPrefix.length);
+        result[paramName] = entry[1];
+      }
+    });
+  }
+  return result;
+};
+
+export const getPathParams = nodes => {
+  const params = {};
+  nodes
+    .filter(n => n.pathParam)
+    .map(n => n.pathParam)
+    .forEach(pp => {
+      params[pp.key.replace(':', '')] = pp.value;
+    });
+  return params;
+};
+
+export const findViewGroup = node => {
+  if (node.viewGroup) {
+    return node.viewGroup;
+  } else if (node.parent) {
+    return findViewGroup(node.parent);
+  }
+};
+const defaultContentViewParamPrefix = '~';
+export const getContentViewParamPrefix = () => {
```

Comment:
defaultContentViewParamPrefix could be placed inside getContentViewParamsPrefix function, since it is only used there.

Reply:
I vote against. Now it's easier to find the constant. It would be even better to move all constants to the up. I can negotiate 😄